testsuite/integration/elytron-oidc-client/src/test/java/org/wildfly/test/integration/elytron/oidc/client/subsystem/SimpleServletWithScope.java

```diff
@@ -0,0 +1,94 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.wildfly.test.integration.elytron.oidc.client.subsystem;
+
+import jakarta.annotation.security.DeclareRoles;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.annotation.HttpConstraint;
+import jakarta.servlet.annotation.ServletSecurity;
+import jakarta.servlet.annotation.WebServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.List;
+import org.wildfly.security.http.oidc.OidcSecurityContext;
+import org.jboss.as.test.integration.security.common.servlets.SimpleServlet;
+
+
+/**
+ * Modified version of {@link SimpleServlet}. Checks token to find claims obtained using scope values in OIDC authentication
+ *
+ * @author Prarthona Paul
+ */
+@DeclareRoles({ SimpleServletWithScope.ALLOWED_ROLE })
+@ServletSecurity(@HttpConstraint(rolesAllowed = { SimpleServletWithScope.ALLOWED_ROLE }))
+@WebServlet(SimpleServletWithScope.SERVLET_PATH)
+public class SimpleServletWithScope extends SimpleServlet{
+
+    /** The serialVersionUID */
+    private static final long serialVersionUID = 1L;
+    public static final String SERVLET_PATH = "/SimpleServletWithScope";
+    public static final String ALLOWED_ROLE = "JBossAdmin";
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        OidcSecurityContext oidcSecurityContext = getOidcSecurityContext(req);
+        final PrintWriter writer = resp.getWriter();
+
+        List<String> scopeClaims = new ArrayList<String>();
+        if (oidcSecurityContext != null) {
+            String scope = oidcSecurityContext.getToken().getClaimValueAsString("scope");
+            displayScopeClaims(scope, scopeClaims, oidcSecurityContext);
+        }
+        writer.write(RESPONSE_BODY);
+        if (scopeClaims.isEmpty()) {
+            writer.write("No additional claims were recovered using the scope values.\n" );
+        } else {
+            writer.write("Claims received using additional scope values: \n" );
+            for (String value : scopeClaims) {
+                writer.write(value);
+            }
+        }
+        writer.close();
+    }
+
+    private OidcSecurityContext getOidcSecurityContext(HttpServletRequest req) {
+        return (OidcSecurityContext) req.getAttribute(OidcSecurityContext.class.getName());
+    }
+
+    private void displayScopeClaims(String scope, List<String> message, OidcSecurityContext oidcSecurityContext) {
```

Comment:
Just a minor comment, since this method retrieves the scope values from the request instead of actually displaying them, maybe we can rename this to `retrieveScopeClaims` or `getScopeClaims` or something like that?

Reply:
changed!